src/sync/event-status-withdrawing-changed.js

```diff
@@ -0,0 +1,112 @@
+const { each, isNull, find } = require('lodash');
+const web3 = require('../web3');
+const { TX_STATUS } = require('../constants');
+const EventSig = require('../config/event-sig');
+const { toLowerCase } = require('../utils');
+const { getTransactionReceipt } = require('../utils/web3-utils');
+const logger = require('../utils/logger');
+const DBHelper = require('../db/db-helper');
+const parseEvent = require('./parsers/multiple-results-event');
+const MultipleResultsEventApi = require('../api/multiple-results-event');
+const EventLeaderboard = require('../models/event-leaderboard');
+const GlobalLeaderboard = require('../models/global-leaderboard');
+
+const getInvestments = async (txs) => {
+  const accumulated = txs.reduce((acc, cur) => {
+    const amount = web3.utils.toBN(cur.amount);
+    if (!cur.betterAddress) cur.betterAddress = cur.centralizedOracleAddress;
+    if (Object.keys(acc).includes(cur.betterAddress)) {
+      acc[cur.betterAddress] = web3.utils.toBN(acc[cur.betterAddress]).add(amount).toString(10);
+    } else {
+      acc[cur.betterAddress] = amount.toString(10);
+    }
+    return acc;
+  }, {});
+
+  return accumulated;
+};
+
+const updateLeaderboardWithdrawing = async (syncPromises, events, limit) => {
+  each(events, async (event) => {
+    syncPromises.push(limit(async () => {
+      try {
+        // get all the participants
+        const bets = await DBHelper.findBet({
+          eventAddress: event.address,
+          txStatus: TX_STATUS.SUCCESS,
+        });
+        const resultSets = await DBHelper.findResultSet({
+          eventAddress: event.address,
+          txStatus: TX_STATUS.SUCCESS,
+          eventRound: 0,
+        });
+        const txs = resultSets.concat(bets);
+        const investments = await getInvestments(txs);
+        const filtered = [];
+        each(txs, (tx) => {
+          if (!find(filtered, {
+            eventAddress: tx.eventAddress,
+            betterAddress: tx.betterAddress,
+          })) {
+            filtered.push(tx);
+          }
+        });
+        // calculate winnings for all participants
+        const promises = [];
+        each(filtered, (tx) => {
+          promises.push(new Promise(async (resolve, reject) => {
+            try {
+              const { eventAddress, betterAddress, resultIndex, centralizedOracleAddress } = tx;
+              const userAddress = betterAddress || centralizedOracleAddress;
+              // calculate winning for this user in this event
+              let winning = '0';
+              if (resultIndex === event.currentResultIndex) {
+                const res = await MultipleResultsEventApi.calculateWinnings({
```

Comment:
we would still have the parallel limit issue here since the `limit()` is only for the outer promise. let me dig into this issue and lets leave this for now.

Reply:
can we move the limit to wrap this part so no inner promises?
We query the bets and resultsets first, then start to limit.

Reply:
Lets use https://caolan.github.io/async/v3/docs.html#eachOfSeries to handle the `calculateWinnings` and keep the outer limit promise. Would prefer not to limit the logic on the inside of an already parallel promise. Not too familiar with the logic of limit to start placing it in async parallel execution. Not too sure what would happen.

This will be slower since its running each calcWinnings individually but this ensures the concurrency limit is not broken.